### ISSUE TITLE
direction CSS property usage

### ISSUE THREAD

**Issue by catalinred** [LeaVerou/css-almanac#38]
I'm interested to find out how many devs use a different `direction` in CSS compared to the `dir` attribute from the markup?

My guess is that the CSS `direction` property is often used for UX things like star rating or similar and has nothing to do with the actual direction of the text in the document.

**Comment by LeaVerou** [comment]
I don’t think this is feasible to do without custom metrics since we don’t have access to the DOM and it would be too expensive to parse the HTML via `DOMParser`, but we could still measure how frequently `dir` is used and on what selectors. @rviscomi correct me if I’m wrong?

**Comment by LeaVerou** [comment]
@catalinred Do you think we should still pursue this by measuring `dir` usage or should we axe it?

**Comment by catalinred** [comment]
If we can measure both ‘dir’ attr from the markup and ‘direction’ property from the CSS, and depending of the resulted numbers, then we could draw some interesting conclusions here.

But if this is too cumbersome and expensive, sure,  we can skip it :)

**Comment by rviscomi** [comment]
@Tiggerito added a [`dir` custom metric](https://github.com/HTTPArchive/legacy.httparchive.org/blob/5f35283f2bb4d9010a5ef802d716af328fca1e49/custom_metrics/markup.js#L112) for the Markup chapter, so this should be possible.

**Comment by LeaVerou** [comment]
So we just measure usage of each dir value?

**Comment by Tiggerito** [comment]
Here's an example of the output for dirs:

```
    "dirs": {
        "html_dir": "ltr",
        "body_nodes_dir": {
            "values": {
                "ltr": 1
            },
            "total": 1
        }
    },
```

So you can report on the values people use for the html dir attribute as well as how often it gets used within the body.

**Comment by LeaVerou** [comment]
One complication is that we don't really know what CSS selectors match. We can guess in certain cases (e.g. `html` or `:root`, but we don't know what e.g. `.foo` is. I could:
* store all distinct CSS selectors that these values are found on, and hope that they'd aggregate nicely over the website corpus, OR
* just measure `dir` values regardless of what selector they come from, OR
* measure `dir` values broken down into 3 categories: a) obviously `<html>`, b) obviously `<body>`, c) any other selector lumped together

Thoughts?

**Comment by catalinred** [comment]
> measure dir values broken down into 3 categories: a) obviously <html>, b) obviously <body>, c) any other selector lumped together

I'd say this breakdown can be helpful.
